pkg/pillar/cmd/verifier/verifier.go

```diff
@@ -500,12 +502,20 @@ func handleGlobalConfigDelete(ctxArg interface{}, key string,
 }
 
 // ImageVerifierFilenames - Returns pendingFilename, verifierFilename, verifiedFilename
-// for the image
-func ImageVerifierFilenames(infile, sha256, tmpID string) (string, string, string) {
+// for the image. The verifierFilename and verifiedFilename always will have an extension
+// of the media-type, e.g. abcdeff112.application-vnd.oci.image.manifest.v1+json
+// This is because we need the media-type to process the blob. Normally, we carry
+// it around in the status (DownloadStatus -> BlobStatus), but those are ephemeral and
+// lost during a reboot. We need that information to be persistent and survive reboot,
+// so we can reconstruct it. Hence, we preserve it in the filename. It is PathEscape'd
+// so it is filename-safe.
+func ImageVerifierFilenames(infile, sha256, tmpID, mediaType string) (string, string, string) {
 	verifierDirname, verifiedDirname := getVerifierDir(), getVerifiedDir()
 	// Handle names which are paths
-	verified := tmpID + "." + sha256
-	return infile, path.Join(verifierDirname, verified), path.Join(verifiedDirname, sha256)
+	mediaTypeSafe := url.PathEscape(mediaType)
```

Comment:
I see the test passing, but I'm confused about how that works if we _change_ the media type here and encode it in the filename. Later, we just read it without any decoding... Especially taking into account that we have pretty complex media types like "application/vnd.docker.distribution.manifest.v2+json""

Reply:
Argh, sorry, we use an unescape function on reading...
Ok, I'll add a unit test to verify the entire chain of encoding-decoding.

Reply:
We recognize that we made a silly mistake and fix it!

Reply:
Take a look now.

Reply:
I'm concerned now about the reliability of my test script, that did not show any problem in 23 iterations the previous night...